### PR TITLE
migrate setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by deronnax**
Just a work in progress for now.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -84,3 +84,112 @@ required-imports = ["from __future__ import annotations"]
 [tool.ruff.pydocstyle]
 # Use Google-style docstrings.
 convention = "google"
+
+[project]
```
Comment (adehad) [inline pyproject.toml:88]: can we put all the project specific config to the top of the file and all the tools at the bottom

**Review by adehad — CHANGES_REQUESTED** [review]
yep no issues on the cloud build failure, small re-ordering but otherwise all good.

Thanks

File: pyproject.toml (hunk 1 above)
Comment (deronnax) [reply pyproject.toml:1]: there you go.

**Review by adehad — APPROVED** [review]

**Comment by adehad** [comment]
Thanks for the contribution!

**Pull request merged**